sql/updates/pre-update.sql

```diff
@@ -26,6 +26,16 @@ DROP TRIGGER IF EXISTS "0_cache_inval" ON _timescaledb_catalog.chunk_constraint;
 DROP TRIGGER IF EXISTS "0_cache_inval" ON _timescaledb_catalog.dimension_slice;
 DROP TRIGGER IF EXISTS "0_cache_inval" ON _timescaledb_catalog.dimension;
 
+DO $$
+DECLARE
+	catalog_version TEXT;
+BEGIN
+  IF EXISTS(SELECT FROM _timescaledb_catalog.metadata WHERE key='timescaledb_version' AND value <> '@UPDATE_FROM_VERSION@') THEN
+		SELECT value INTO catalog_version FROM _timescaledb_catalog.metadata WHERE key='timescaledb_version';
+		RAISE EXCEPTION 'catalog version mismatch, expected "%" seen "%"', '@UPDATE_FROM_VERSION@', catalog_version;
+  END IF;
+END$$;
```

Comment:
```suggestion
DO $$
DECLARE
    catalog_version TEXT;
BEGIN
    SELECT value INTO catalog_version FROM _timescaledb_catalog.metadata WHERE key='timescaledb_version' AND value <> '@UPDATE_FROM_VERSION@';
    IF FOUND THEN
        RAISE EXCEPTION 'catalog version mismatch, expected "%" seen "%"', '@UPDATE_FROM_VERSION@', catalog_version;
    END IF;
END$$;
```

Don't need two SELECT queries over the same metadata table.